python/private/pypi/extension.bzl

```diff
@@ -640,17 +795,77 @@ def _pip_impl(module_ctx):
                 for key, values in whl_map.items()
             },
             packages = mods.exposed_packages.get(hub_name, []),
+            platform_constraint_values = mods.platform_constraint_values.get(hub_name, {}),
             groups = mods.hub_group_map.get(hub_name),
         )
 
     if bazel_features.external_deps.extension_metadata_has_reproducible:
         # NOTE @aignas 2025-04-15: this is set to be reproducible, because the
         # results after calling the PyPI index should be reproducible on each
         # machine.
-        return module_ctx.extension_metadata(reproducible = True)
+        return module_ctx.extension_metadata(reproducible = False)
     else:
         return None
 
+_default_attrs = {
+    "arch_name": attr.string(
+        doc = """\
+The CPU architecture name to be used.
+
+:::{note}
+Either this or {attr}`env_platform_machine` should be specified.
+:::
+""",
+    ),
+    "constraint_values": attr.label_list(
```

Comment:
I think this should be a string_list, not label list?

If it's a label, it'll trigger repo-evaluation, even if it goes unused.

Reply:
I thought it does cause repo fetching? Though, when I think back, maybe I'm confusing label_list() in repo/bzlmod with `rctx.path(<some label>)`.

In any case, lets double check to verify.